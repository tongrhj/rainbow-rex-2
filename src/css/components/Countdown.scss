@import '../variables';

.countdown {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 4px;
}

.countdown .remaining {
  background-color: $countdown-background;
  height: 5px;
  box-shadow: 0 1px 5px $countdown-shadow;
  transition: width 0.05s;
}
