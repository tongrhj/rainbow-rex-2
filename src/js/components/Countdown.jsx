import countdownCss from '../../css/components/Countdown.scss'
import React from 'react'

export default ({timeLeft, totalTime}) => {
  timeLeft = totalTime ? timeLeft / totalTime : 0
  return (
    <div className={countdownCss.countdown}>
      <div className={countdownCss.remaining}
        style={{ width: `${timeLeft * 100}%` }}
      />
    </div>
  )
}
