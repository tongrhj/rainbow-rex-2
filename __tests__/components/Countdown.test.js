import React from 'react'
import Countdown from '../../src/js/components/Countdown'
import renderer from 'react-test-renderer'

it('renders correctly with 0 width', () => {
  const tree = renderer.create(
    <Countdown />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('changes the width when timeLeft / totalTime changes', () => {
  const component = renderer.create(
    <Countdown timeLeft={1000} totalTime={1000} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  const component2 = renderer.create(
    <Countdown timeLeft={500} totalTime={1000} />
  )
  tree = component2.toJSON()
  expect(tree).toMatchSnapshot()

  const component3 = renderer.create(
    <Countdown timeLeft={0} totalTime={1000} />
  )
  tree = component3.toJSON()
  expect(tree).toMatchSnapshot()
})
