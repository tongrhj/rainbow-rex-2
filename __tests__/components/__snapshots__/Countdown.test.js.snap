exports[`test changes the width when timeLeft / totalTime changes 1`] = `
<div
  className="countdown">
  <div
    className="remaining"
    style={
      Object {
        "width": "100%",
      }
    } />
</div>
`;

exports[`test changes the width when timeLeft / totalTime changes 2`] = `
<div
  className="countdown">
  <div
    className="remaining"
    style={
      Object {
        "width": "50%",
      }
    } />
</div>
`;

exports[`test changes the width when timeLeft / totalTime changes 3`] = `
<div
  className="countdown">
  <div
    className="remaining"
    style={
      Object {
        "width": "0%",
      }
    } />
</div>
`;

exports[`test renders correctly with 0 width 1`] = `
<div
  className="countdown">
  <div
    className="remaining"
    style={
      Object {
        "width": "0%",
      }
    } />
</div>
`;
